cupyx/fallback_mode/fallback.py

```diff
@@ -17,10 +20,37 @@ class _RecursiveAttr:
     when user calls fallback_mode. numpy is an instance of this class.
     """
 
-    def __init__(self, numpy_object, cupy_object):
+    def __init__(self, numpy_object, cupy_object, array=None):
+        """
+        _RecursiveAttr initializer.
+
+        Args:
+            numpy_object (method): NumPy method.
+            cupy_method (method): Corresponding CuPy method.
+            array (ndarray): Acts as flag to know if _RecursiveAttr object
+            is called from ``ndarray`` class. Also, acts as container for
+            modifying args in case it is called from ``ndarray``.
+            None otherwise.
+        """
 
         self._numpy_object = numpy_object
         self._cupy_object = cupy_object
+        self._fallback_array = array
+
+    def __instancecheck__(self, instance):
```

Comment:
This method is not needed now.

Reply:
Hmm...I'm not sure. Aren't there any other attributes which can make use of `isinstance` besides `dtypes` and `ndarray`. Oh, `isinstance(vectorized_obj, numpy.vectorize)` won't work without it. Maybe there are some other cases.
I think it's safe to only removing following from `__instancecheck__`:
```python
if self._cupy_object is cp.ndarray:
    return isinstance(instance, ndarray)
```

Reply:
I see. Why not adding the test checking it?

Reply:
Done, added test for a dtype, `vectorized_object` and `datatime64_object`.